libknet/compress.c

```diff
@@ -209,6 +209,28 @@ static int compress_lib_test(knet_handle_t knet_h)
 		log_err(knet_h, KNET_SUB_COMPRESS, "Unable to compress test buffer. Please check your compression settings: %s", strerror(savederrno));
 		errno = savederrno;
 		return -1;
+	} else if ((long unsigned int)dst_comp_len >= KNET_DATABUFSIZE) {
+		/*
+		 * compress not effect, try again using default compression level
+		 */
+		log_warn(knet_h, KNET_SUB_COMPRESS, "Useless compression level: %d", knet_h->compress_level);
```

Comment:
I would prefer a more meaningful message here.

/*
 * compress not effective, try again using default compression level when available
 */
log_warn( .... "Requested compression level (%d) did not generate any compressed data (source: %zu destination: %zu)".. and add source and destination length to the message for debugging purposes.